packages/babel-helpers/src/helpers/iterableToArrayLimitLoose.ts

```diff
@@ -0,0 +1,20 @@
+/* @minVersion 7.0.0-beta.0 */
+
+export default function _iterableToArrayLimitLoose<T>(
+  arr: Array<T>,
+  i: number,
+) {
+  var iterator: Iterator<T> & Function =
+    arr &&
+    ((typeof Symbol !== "undefined" && arr[Symbol.iterator]) ||
+      (arr as any)["@@iterator"]);
+  if (iterator == null) return;
+
+  var _arr: T[] = [];
+  var step;
+  iterator = iterator.call(arr);
+  while (arr.length < i && !(step = iterator.next()).done) {
```

Comment:
The function argument `arr` accepts `Array<T>` here as we are accessing the `.length` property, which doesn't exist on normal `Iterator`s.

Reply:
I suppose this would include `String` as well. I'll update the `arr` arg to accept `Array<T> | String`

Reply:
Ughhh I'm quite confident that this is a bug, and we should be using `_arr` instead of `arr`. Can you mark `arr` as `Iterable<T>`, and add a `@ts-expect-error` here? I'll open an issue to track it.

Reply:
Oh whaat, this helper (and `slicedToArrayLoose`) have never been used. They have been introduced in https://github.com/babel/babel/commit/ae7d5367f1c3d438667242d6925db024f875fccd, but there has never been any code actually loading them.

It doesn't matter how they are converted to TS at this point -- could you instead open a separate PR to delete them? 😅